From 53b5286b50afbdce92f42397517e01937c8f875d Mon Sep 17 00:00:00 2001
From: Steve Nogar <snogar@gmail.com>
Date: Sun, 1 Mar 2020 13:01:34 -0500
Subject: [PATCH] fix mac trying to use epoll instead of kqueue

macOS was incorrectly attempting to use epoll/poll instead of kqueue, occassionally generating errors such as RuntimeError: concurrent poll() invocation and breaking python nodes.
---
 clients/rospy/src/rospy/topics.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git clients/rospy/src/rospy/topics.py clients/rospy/src/rospy/topics.py
index 124864cfa..3ef4412ae 100644
--- clients/rospy/src/rospy/topics.py
+++ clients/rospy/src/rospy/topics.py
@@ -192,7 +192,13 @@ class Poller(object):
     on multiple platforms.  NOT thread-safe.
     """
     def __init__(self):
-        if hasattr(select, 'epoll'):
+        if hasattr(select, 'kqueue'):
+            self.poller = select.kqueue()
+            self.add_fd = self.add_kqueue
+            self.remove_fd = self.remove_kqueue
+            self.error_iter = self.error_kqueue_iter
+            self.kevents = []
+        elif hasattr(select, 'epoll'):
             self.poller = select.epoll()
             self.add_fd = self.add_epoll
             self.remove_fd = self.remove_epoll
@@ -202,12 +208,6 @@ class Poller(object):
             self.add_fd = self.add_poll
             self.remove_fd = self.remove_poll
             self.error_iter = self.error_poll_iter
-        elif hasattr(select, 'kqueue'):
-            self.poller = select.kqueue()
-            self.add_fd = self.add_kqueue
-            self.remove_fd = self.remove_kqueue
-            self.error_iter = self.error_kqueue_iter
-            self.kevents = []
         else:
             #TODO: non-Noop impl for Windows
             self.poller = self.noop
-- 
2.25.0

